### PR TITLE
cadl emitter ci and release pipeline

### PR DESCRIPTION
**Pull request by chunyu3**
# Description
Fix https://github.com/Azure/autorest.csharp/issues/2502

- cadl csharp emitter ci and release pipeline

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: eng/CaldEmitterUT.ps1 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#Requires -Version 7.0
```
Comment (pshao25) [inline eng/CaldEmitterUT.ps1:1]: CadlEmitterUT

Comment (chunyu3) [reply eng/CaldEmitterUT.ps1]: updated

File: eng/CadlEmitterUT.ps1 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#Requires -Version 7.0
```
Comment (lirenhe) [inline eng/CadlEmitterUT.ps1:1]: Can we call this file  ExecuteCadlEmitterUnitTests which is more consistent with the existing script

File: eng/pipelines/cadl-ci.yml [hunk 3]
```diff
@@ -22,33 +57,76 @@ stages:
...
-            npm install
-            npm run build
-            npm pack
-          displayName: 'Install package'
-          workingDirectory: $(Build.SourcesDirectory)/autorest.csharp/src/CADL.Extension/Emitter.Csharp
-        - pwsh: .\eng\CadlGenerate.ps1
+            command: custom
+            customCommand: pack
+            workingDir: $(Build.SourcesDirectory)/src/CADL.Extension/Emitter.Csharp
+        - pwsh: ./eng/CadlEmitterUT.ps1
```
Comment (lirenhe) [inline eng/pipelines/cadl-ci.yml:79]: Will this step use the Emitter.csharp that is built in the previous step?

Comment (chunyu3) [reply eng/pipelines/cadl-ci.yml:79]: yes

File: eng/CadlEmitterUT.ps1 (hunk 2 above)
Comment (chunyu3) [reply eng/CadlEmitterUT.ps1:1]: updated

**Review by lirenhe — APPROVED** [review]

**Pull request merged**